src/tls13.c

```diff
@@ -11805,10 +11813,10 @@ static int SendTls13NewSessionTicket(WOLFSSL* ssl)
 {
     byte*  output;
     int    ret;
+    word32 length;
     int    sendSz;
     word16 extSz;
-    word32 length;
-    word32 idx = RECORD_HEADER_SZ + HANDSHAKE_HEADER_SZ;
+    word16 idx = RECORD_HEADER_SZ + HANDSHAKE_HEADER_SZ;
```

Comment:
Seems like the wrong idea to use `word16` for `idx` -- makes it more vulnerable to overflow.  Though to be fair, there are currently no checks in the body of the function for overflow.  Perhaps revert `idx` to `word32`, and add casts where necessary, immediately preceded by overflow tests?

Reply:
Ah, yes. Good point.` word32` for `idx`. What would you use for an error code? Add something like this in [here tls13.c](https://github.com/wolfSSL/wolfssl/blob/ae0d73d9fdde9c4d633283dcda6386e8743d21e0/src/tls13.c#L11936)?

```
    if (idx > WOLFSSL_MAX_16BIT) {
        return BAD_LENGTH_E;
    }
```

Reply:
Although the prior change seemed happy with the VS/Windows build, today I was seeing the "_conversion from 'word32' to 'word16', possible loss of data_" error again today. 

So in addition to removing the `ASN_IMPLICIT_TAG_MASK` in `ssl.asn1.c`, I've also added explicit `(word16)` type casts to the 32-bit `idx` in the `tls13.c` file.